IBPSA/Utilities/IO/Reports/CSVWriter.mo

```diff
@@ -46,14 +68,22 @@ equation
 
 algorithm
   when sampleTrigger then
-    str :=String(time) + delimiter;
+    str :=String(time) + _delimiter;
     for i in 1:nin-1 loop
-      str :=str + String(u[i]) + delimiter;
+      str :=str + String(u[i]) + _delimiter;
     end for;
     str :=str + String(u[nin]);
     Modelica.Utilities.Streams.print(str, fileName);
+    dataLinesWritten :=dataLinesWritten + 1;
   end when;
 
+  // now that the file has been written completely, prepend the header
```

Comment:
Please put this functionality in the `destructor` of `FileWriter` as `terminal()` is not called when simulating an FMU, e.g., in JModelica (because the FMU does not know that it needs to terminate). If you put it into the `destructor` then it works.

Reply:
great, thanks!

Reply:
I started implementing this. At this point I'm a bit stuck since I'm not sure whether it is possible to pass a string/double argument to a Modelica object destructor? If that's not possible then I cannot pass to the constructor how many rows have been printed. The alternative is to then to 1) store this information in the file writer object itself when calling a function that writes a line (but then we can't use `Modelica.Utilities.Streams.print`) or 2) read from the file itself how many lines exist. We then have to subtract the header lines.

Reply:
suggested to use 1) during IBPSA call